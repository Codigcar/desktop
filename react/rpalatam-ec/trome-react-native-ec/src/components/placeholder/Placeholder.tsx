import React, { useMemo } from 'react'
import { SvgProps } from 'react-native-svg'

import DeporPlaceholder from '../../assets/brands/depor/placeholder.svg'
import ElComercioPlaceholder from '../../assets/brands/elcomercio/placeholder.svg'
import GestionPlaceholder from '../../assets/brands/gestion/placeholder.svg'
import Peru21Placeholder from '../../assets/brands/peru21/placeholder.svg'
import TromePlaceholder from '../../assets/brands/trome/placeholder.svg'
import { App } from '../../utils/config'
import Box from '../box'

type Props = {
  width?: number
}

const Placeholder: React.FC<Props> = ({ width = 100 }) => {
  const svgProps: SvgProps = useMemo(() => {
    return {
      width,
      height: width * 0.8,
      fill: '#000',
      fillOpacity: 0.2,
    }
  }, [width])

  const Isotipo = useMemo(() => {
    return App.select({
      depor: <DeporPlaceholder {...svgProps} />,
      elcomercio: <ElComercioPlaceholder {...svgProps} />,
      gestion: <GestionPlaceholder {...svgProps} />,
      peru21: <Peru21Placeholder {...svgProps} />,
      trome: <TromePlaceholder {...svgProps} />,
    })
  }, [svgProps])

  return (
    <Box.View
      bg="trueGray-200"
      width="100%"
      height="100%"
      alignItems="center"
      justifyContent="center">
      {Isotipo}
    </Box.View>
  )
}

export default Placeholder
