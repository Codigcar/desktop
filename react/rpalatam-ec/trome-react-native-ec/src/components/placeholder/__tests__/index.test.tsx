import { act, render } from '@testing-utils/library'
import React from 'react'

import Placeholder from '../Placeholder'

describe('Placeholder component', () => {
  test('should match with her snapshot', async () => {
    const { toJSON } = render(<Placeholder />)
    await act(async () => {
      expect(toJSON()).toMatchSnapshot()
    })
  })
})
