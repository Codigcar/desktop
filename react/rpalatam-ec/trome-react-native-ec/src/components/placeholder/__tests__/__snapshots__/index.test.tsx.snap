// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder component should match with her snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#E5E5E5",
          "display": "flex",
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <SvgMock
      fill="#000"
      fillOpacity={0.2}
      height={80}
      width={100}
    />
  </View>
</RNCSafeAreaProvider>
`;
